### PR TITLE
add a test for new binary artifact download concurrent download limit

### PR DESCRIPTION
**Pull request by tomerd**
motivation: recent change to SwiftPM introduce a limit on # of concurrent binary artofacts downloads, but was missing a test

changes: add a test for the new functionality

### CODE REVIEW

File: Sources/SPMTestSupport/MockWorkspace.swift [hunk 1]
```diff
@@ -398,12 +398,20 @@ public final class MockWorkspace {
...
+                observabilityScope: observability.topScope
+            )
+            result(graph, observability.diagnostics)
+        } catch {
+            // helpful when graph fails to load
+            if observability.hasErrorDiagnostics {
+                print(observability.diagnostics.map{ $0.description }.joined(separator: "\n"))
+            }
+            throw error
+        }
```
Comment (tomerd) [inline Sources/SPMTestSupport/MockWorkspace.swift:414]: this is not directly related but generally helps when writing tests

**Comment by tomerd** [comment]
@swift-ci please smoke test

cc @plu

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

**Comment by plu** [comment]
Thank you for adding tests, and sorry I didn't do it in the first place.
